svm/src/transaction_processor.rs

```diff
@@ -558,8 +726,19 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
         error_counters: &mut TransactionErrorMetrics,
         log_messages_bytes_limit: Option<usize>,
         program_cache_for_tx_batch: &ProgramCacheForTxBatch,
+        unique_loaded_accounts: &mut HashMap<Pubkey, AccountSharedData>,
+        dedup_nonce_lookup: &mut HashSet<Hash>,
     ) -> TransactionExecutionResult {
-        let transaction_accounts = std::mem::take(&mut loaded_transaction.accounts);
+        let mut transaction_accounts = std::mem::take(&mut loaded_transaction.accounts);
+
+        // get latest value of acct struct from unique_loaded_accounts
+        for (key, acct) in transaction_accounts.iter_mut() {
```

Comment:
this loop effectively obsoletes the existence of the `loaded_transaction.accounts` field, right?

i think parallel existence of `loaded_transaction.accounts` and `unique_loaded_accounts` with strong coherency rule could be a maintenance burden.

instead, how about changing type of `LoadedTransaction::accounts` to `Vec<LoadedUniqueAccount>` and `type LoadedUniqueAccount = Rc<RefCell<AccountSharedData>>`?